packages/node_modules/nav-frontend-stegindikator/src/stegindikator.tsx

```diff
@@ -0,0 +1,237 @@
+
+    constructor(props: StegindikatorProps) {
+        super(props);
+
+        let initialAktivtSteg;
+        if (this.props.aktivtSteg !== undefined) {
+            initialAktivtSteg = this.props.aktivtSteg;
+        } else {
+            initialAktivtSteg = this.getDefaultActiveStegIndex();
+        }
+
+        this.state = {
+            aktivtSteg: initialAktivtSteg,
+            visLabel: props.visLabel,
+            kompakt: props.kompakt
+        };
+
+        if (props.autoResponsiv) {
+            window.addEventListener('resize', this.adjustSize);
+        }
+    }
+
+    componentDidMount() {
+        if (this.props.autoResponsiv) {
+            this.adjustSize();
+        }
+    }
+
+    componentWillReceiveProps(nextProps: StegindikatorProps) {
+        if (!this.props.autoResponsiv && nextProps.autoResponsiv) {
+            window.addEventListener('resize', this.adjustSize);
+        }
+
+        if (nextProps.autoResponsiv) {
+            this.setState({
+                visLabel: (this.canShowLabel()) ? nextProps.visLabel : false,
+                kompakt: nextProps.kompakt
+            });
+        } else {
+            this.setState({
+                visLabel: nextProps.visLabel,
+                kompakt: nextProps.kompakt
+            });
+        }
+
+        if (nextProps.aktivtSteg !== undefined && nextProps.aktivtSteg !== this.state.aktivtSteg) {
+            this.setState({
+                aktivtSteg: nextProps.aktivtSteg
+            });
+        }
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('resize', this.adjustSize);
+    }
+
+    handleClick = (e, index) => {
+        e.preventDefault();
+        if (!this.props.onBeforeChange || this.props.onBeforeChange(index)) {
+            this.setState({
+                aktivtSteg: index
+            });
+            if (typeof this.props.onChange === 'function') this.props.onChange(index)!;
+        }
+    }
+
+    getDefaultActiveStegIndex() {
+        let index;
+        if (this.props.children) {
+            React.Children.forEach(this.props.children, (child, i) => {
+                if (React.isValidElement(child)) {
+                    const clone = React.cloneElement(child as React.ReactElement<any>);
+                    if (clone.props['aktiv']) {
+                        index = i;
+                    }
+                }
+            });
+        } else {
+            index = this.props.steg.findIndex((steg) => !!steg.aktiv);
+        }
+        return (index !== -1) ? index : 0 ;
+    }
+
+    getNumSteg() {
+        if (this.props.children) {
+            return React.Children.toArray(this.props.children).filter((child) => React.isValidElement(child)).length;
+        }
+        return this.props.steg.length;
+    }
+
+    getDimensions() {
+        const numSteg = this.getNumSteg();
+        const remSize = parseFloat(String(getComputedStyle(document.documentElement).fontSize));
+        const margin = remSize * 1.25;
```

Comment:
Kan ikke det meste her styres med css?

Reply:
Hm, tenker det kan være greit å ta høyde for tilfeller der `1rem` ikke er `16px`?